docs/install.md

```diff
@@ -0,0 +1,150 @@
+variable to `SIM`:
+
+```bash
+export SGX_MODE=SIM
+```
+
+SGX hardware mode uses capabilities for trusted execution in the
+processor to protect confidentiality and integrity of computation. SGX
+hardware mode requires processor support for SGX
+([commonly available on recent Intel processors](https://ark.intel.com/content/www/us/en/ark/search/featurefilter.html)).
+To use SGX in hardware mode set the `SGX_MODE` environment variable to
+`HW`:
+
+```bash
+export SGX_MODE=HW
+```
+
+The remainder of this section provides information about preparing to
+run Private Data Objects using SGX in hardware mode. Specifically, there
+are steps that must be taken to enable attestation of the hardware
+platform using the Intel Attestation Service (IAS).
+
+### Update the BIOS
+
+It may be advisable to update your BIOS to pick up any recent
+patches. For example, if you want to run PDO on an Intel NUC, you can
+visit the
+[Intel Download Center](https://downloadcenter.intel.com/)
+(and select `Mini PCs`) to find the latest BIOS release. Other vendors
+may provide similar updates.
+
+Note: If the Trusted Computing Base (TCB) of the platform performing an
+attestation is outdated, IAS can still verify an attestation; however,
+the verification may include a `GROUP_OUT_OF_DATE` or
+`CONFIGURATION_NEEDED` quote status. It is up to the verifier to decide
+whether to trust an attestation from a platform whose TCB has been
+identified as outdated. Note that PDO will flag this situation with a
+warning.
+
+### Create an IAS Authentication Key
+
+Access to IAS requires that a service provider create an identity (SPID)
+and a client authentication key. To create the SPID and the key you
+first need to register with the
+[Intel Developer Portal](https://api.portal.trustedservices.intel.com/developer).
+
+Once your registration is complete subscribe for a
+[linkable quote](https://api.portal.trustedservices.intel.com/EPID-attestation)
+to create the client authentication key. The key will be available from
+your profile page.
+
+Assuming that existing keys are found in `${PDO_SGX_KEY_ROOT}` (which
+defaults to the directory
+`${PDO_SOURCE_ROOT}/build/keys/sgx_simulation`),
+copy the key from your profile page into a file called
+`${PDO_SGX_KEY_ROOT}/sgx_spid_api_key.txt`.
+
+In addition, your SPID can be found on your developer portal profile
+page. Copy the contents of the SPID into the file
+`${PDO_SGX_KEY_ROOT}/sgx_spid.txt`.
+
+### Install the SGX Kernel Driver (Hardware Support)
+
+SGX can run in either simulation or hardware mode. No kernel driver is
+required to run in simulation mode. However, if you plan to run with SGX
+hardware support, it is necessary to install the SGX kernel driver. The
+following commands will download and install v2.4 of the SGX kernel
+driver:
+
+```bash
+wget https://download.01.org/intel-sgx/linux-2.4/ubuntu18.04-server/sgx_linux_x64_driver_778dd1f.bin
+sudo /bin/bash sgx_linux_x64_driver_778dd1f.bin
+```
+
```

Comment:
it's probably worth mentioning that the driver is bound to the kernel version.
So when the kernel is updated and changes version, the driver has to be reinstalled.

Reply:
i would prefer to stay out of sysadmin 101. we cannot (and i have no intention of attempting) cover all possible cases.

Reply:
> i would prefer to stay out of sysadmin 101. we cannot (and i have no intention of attempting) cover all possible cases.

It's very non-obvious that the installer only installs the driver for the current kernel version and the error after a kernel update is not thaat obvious, so probably rather sysadmin 102 rather than 101 :-) 
[as soon as we go to DCAP this should be a moot issue though as in DCAP (finally) adopted DKMS and the module is automatically recompiled during kernel updates ...]